logstreams/src/test/java/io/zeebe/logstreams/state/StateSnapshotControllerTest.java

```diff
@@ -73,14 +74,14 @@ public void shouldTakeSnapshot() throws Exception {
   }
 
   @Test
-  public void shouldOpenNewDatabaseIfNoSnapshotsToRecoverFrom() throws Exception {
+  public void shouldDoNothingIfNoSnapshotsToRecoverFrom() throws Exception {
     // given
 
     // when
-    final long lowerBoundSnapshotPosition = snapshotController.recover();
+    snapshotController.recover();
```

Comment:
Should we fail then? I'm wondering what we should do in the processor actually, if we have no snapshot to open (because they are corrupted)

Reply:
This is probably a different issue - current behavior is not to fail. If we have no snapshots at all, then :heavy_check_mark:  If we had some corrupted snapshots, then probably we should step down and throw away our state, but that's a bigger issue, I guess

Reply:
Had a look in the code - if the snapshot was corrupt and the last one we throw an exception, where the processor goes into failed state then. so should be fine